src/ws/cockpitauth.c

```diff
@@ -944,14 +943,10 @@ build_session_credentials (CockpitAuth *self,
     {
       raw = cockpit_authorize_parse_basic (authorization, &user);
 
-      /* If "password" is in the X-Authorize header or X-Superuser is
-         not "none" (and we are not root), we keep the password.
+      /* If X-Superuser is not "none" (and we are not root), we keep
```

Comment:
Is X-Authorize now completely replaced by X-Superuser?

Reply:
Yes, although "replaced" is not exactly right.  It went in steps:

- Historically, cockpit-ws would store the login password and use it for any number of things, like starting sudo when the first superuser channel was opened, and connecting to remote machines when the first channel to a remote machine was opened. Cockpit-ws would remember the password for the duration of the whole session. The user could opt out of this with the checkbox on the login page.  I don't think many people ever understood what this checkbox actually does.

- Then we changed how sudo is started.  When it happens in the middle of a session, cockpit-ws is not involved at all, but we wanted to keep the convenience of re-using the login password for sudo during login. So the login page tells cockpit-ws wether sudo should be started via the X-Superuser header (this is remembered from the last session), and cockpit-ws passes that on to the bridge, which will start sudo as part of its initialzation. We didn't want to introduce new places with passwords in the protocol, so we kept using the "authorize" messages for that. But now we know when cockpit-ws can forget the password.  Except it was still used for SSH connections, so we kept the checkbox for that, and the X-Authorize header that goes with it.

- In this PR, we stop using the login password for remote machines completely, and cockpit-ws now only needs to remember the password for the superuser startup. The checkbox and its X-Authorize header is gone.